### ISSUE TITLE
Stability First. As a speler/SL I want to be able to use our awesome IT-systems

### ISSUE THREAD

**Issue by WafflesMcDuff** [eosfrontier/no_repo_issues#9]
TL;DR:
Let's make our systems in a way that they will run with AND without internet. For example strati.os: it we havo no internet we can still look at the last known updates. As soon as internet is available it will sync.

Long version:
Our IT-systems are amazing, but in the last 3 events a lot of the systems were often down because of internet problems. There are different things affecting these problems, from no acces to the internet, to not being able to reconnect to the wifi, to internet having too many users on it. This is a shame, because I've heard a lot of people giving up on the awesome systems that you guys build. Also, this causes strain on especially the figuranten-team that is already overloaded with work.

I propose to focus on stabilizing our systems before we build new stuff. Quality over quantity. My personal idea would be to run systems locally first an application specificdatabase. This database is connected with an api to a database in the cloud. When this database gets new data a streamset will trigger and the local database gets updated. We also could store the data of our last change, so people can read how up to date the screen is that they're looking at.

Some systems, such as the namegenarator, might not even need an active data connection at all times. Running these in an offline mode will releave the internet of these connections. Also, this will create some trust among SL's, spelers and figuranten about the IT system.

With the permission of Elmar I want to share this:
At one point at the last event I took a break in the figuranten-huis. Figuranten were preparing for roles, and the porto was continuously active with all SL's asking for figuranten (as it was most of the time during the event). Figuranten were asking for stuff, like the namegenerator. But the internet stopped working some hours ago. I saw Elmar sit at the computer, and he was just lost, staring at the computer, and playing the dinosaur-game you see when you don't have any internet connection. I just remember that he looked at me with a look in his eyes that told me that he really tried, but he lost hope. He didn't feel the support, and this was because of the unavailability of the internet for the third event in a row.

We need to fix this guys, I think it should be our top-priority.

**Comment by WafflesMcDuff** [comment]
@HesterdeTester Apr 8, 2019 at 4:53 AM
Had a quick conversation with @WafflesMcDuff on discord where we were able to get on the same page again and understand each other better. I think your comment really helped us, @SulpherStaer :) :) :)

I like the idea of systems running like this:
Normal: runs on through our network on the internet
No internet: runs through our network on our offline server
No network: runs in a read-ready version of our app

Also, I do agree with @sanderburger, step by step would be smart. You sketched the whole situation well, and this is a complicated project we need to align on. For example, I do have some test-related questions like: what information is leading when we have differences in the data-bases? Is there functionality that we still need to limit when we're in no network mode (think of systems that handle IC-cash)?
Refinement needed.

---
@SanderBurger Apr 8, 2019 at 4:25 AM (edited)
@SulpherStaer It is important to note here that we do not rely solely on online. It is a weakness, however. Hence my point that we need to be clear about what issues we are fixing and in what way.

I understand the need to so something NOW, but if we shoot off in various directions trying to fix various issues in various versions of what we perceive to the the main problem we'll get in a lot of trouble down the road and decrease the reliability instead of increasing it.

Reliable and dependable tech is the goal. A form of offline ability, local network and/or automated fallback is the means. Which type we will have to decide in the next few weeks. I saw you already picking stuff up, and it is awesome that you are already looking at possibilities. That will definitively increase the number of solutions we have.

I know step by step is not a sexy approach, but I prefer all of us looking back on this once we have cracked with with a clear ' we made the right call '.

---
@SulpherStaer Apr 7, 2019 at 6:30 PM (edited)
"One possible solution to this is to join all our laptops to the eosfrontier active directory domain and thus using AD-integrated DNS becomes possible for much of our architecture. However I’m loathe to do that because then NEXUS becomes a single point of failure."

"However, I think a big part of the problem is that we (the IT Team) don't know what specifically is broken"

The strong reliance on ONLINE Tech-bare systemen is our single point of failure already. And that is what is broken.
If internet for whatever reason is being freaky, for whatever reason, a lot of systems go down very quickly.

Things need to be available offline, without connectivity. We have machines, if a machine for whatever reason, loses connection to WiFi, or internet, or whatever connection, the tech-bare systemen should still be available.

I am aware that this will cause a LOT of issues with data-syncing, but it is something that I strongly believe is a MUST. The syncing issue is in my eyes, a headache for us at the backend, but at least not for our front end users.

I know that I am going to see if I can get the NameGenerator offline, or atleast still running local cached when it's downloaded. Because unavailability for such a relatively simple system, is not acceptable.

Techbaar =/= Online. We need offline availability, for all game important systems. I am aware that some are impossible, or the headache becomes waaay to big (data syncing). But many things things should be able to work on a single offline machine such as Name Generator, Medicorum, Weapon EMS, Surgical Dice Roller. Two of these relly on databases, but are only run on ONE device. We should be able to do this.


---
@SanderBurger Apr 7, 2019 at 4:25 PM (edited)
Alright, since the WoS is over, my thoughts. It has been an issue that has been vexing me for quite a while, and one that I would dearly wish to solve. A significant amount of time, effort, resources and thoughts have been poored into it, and while I am happy with the improvements, ol' Murphy has graced us with his presence more then once. I am in favor of picking this up as a high priority, with the following additions, thoughts and notes.

So far we have not been able to comprehensively crack it. This sucks. We get it better every time, but this has not been super-good for morale in certain areas. That having been said, we've had rotten luck. We'll need buy-in from the entire orga-team. Trust has to go both ways. If this is absent, see my last point.

While I appriciate focus, stopping all new stuff would be counter-productive. This is due to:

The voluntary nature of the team - we do need our passion projects / cookies.
Available expertise - network and connectivity is not something all members of the time can help with.
Expectations - the old '9 woman can't produce 1 baby in 1 month' addage - this may be a long term project that runs several years. The problem is complex, the required resources may reqiure investment beyond the scope of a single event and ready-made solutions for this problem unavailable.

Instead of I would propose a taskforce that picks that as their major project for this coming period. That leads me to the second one.

We have to clearly specify the questions and solutions that will get us to a 'minimum viable Eos', aka what do we need to run the necessary systems without interruption. This will include a fair bit of gathering hard data, stripped of anecdotal stuff - there is a lot of frustration going on, and it can cause us to focus resources and goodwill on the wrong things. I know this will be hard, but we can't place the full burden for everything in any one place or on any one person.

Whatever solution we come up with will have to be tested as well, preferably under something approximating game conditions. This is something the IT team cannot do alone.

Solutions have to be in reach and supportable. This requires looking at the solutionspace that we have on hand - new stuff is awesome, but will quickly limit available resources and flexibility of the team. Getting together and getting a practical up and running would be my proposal, and an honest assessment what we can actually do. If not, see the final point.

A partial issue is that we have mechanisms in place to mitigate ( like the full drive backup on every system ), but knowledge of these systems in the field is inconsistent. I would therefore like to renew the search for a person who can support the team during gametime that is not an sl or player, to be added to the IT team.

Lastly, in extremis we'll have to consider the ceiling of what we can actually offer, underpromise and overdeliver. This may mean taking a step back and taking a hard look at what we can actually guarantee. This won't be fun, but if we have to guarantee a limited scope and that works, we have successes we can build on. Limitations we can communicatie, dissapointment is harder to mitigate.

---
@WafflesMcDuff Apr 5, 2019 at 12:01 PM
One thing to keep in mind, is that FXI was very much a "let's grow together" event. Prior to the event we had not tried to run our gear on their infrastructure, did not have a clear picture of where the WiFi was strong/weak, nor did we have a picture of where all the physical network jacks were. We have all that info now and can build on it for FXII

---
@WafflesMcDuff Apr 5, 2019 at 11:59 AM (edited)
@hesterleahdestoppelaar I'm definitely not dimissing you. Not even a little. I want to understand what is happening. However, I think a big part of the problem is that we (the IT Team) don't know what specifically is broken. The old location had AWFUL internet connection. The new location is MUCH better. (Although we had signal problems in a few key areas in the Bastion which we already have planned purchases to fix).

To the best of my knowledge (which could be VERY flawed) internet in the figurhuis never went down this event. The google drive should have been 100% accessible, and no one told me otherwise. Add to that the fact that all the laptops have the google drive 100% cached to them before the event. Perhaps it's a training/knowledge sharing/communication issue (like when we weren't clear telling you the site was fully available)? I'm not sure.

Like I said in my first comment, we can discuss this in greater detail and break it down into each of the components and suggested strategies when I’m back from Italy.

---
@HesterdeTester Apr 5, 2019 at 7:18 AM (edited)
I want to discuss this in more detail in person or via voice, because I think it will create more understanding on both sides. I know that we had another problem with the site being down last weekend, but that problem was fixed quickly with the DSN-solution (for which I'm really gratefull).

The problem I'm talking about is different: it's about the unavailability of internet. Both the internet from the location and our own network have failed the last event. I could not reach the google drive for several hours, and I had to tell people they had to wait. Elmar had to tell the figuranten that even after the DNS-fix he could not help them.

From your message I don't get the feeling you understand this. I feel like you're telling me that I'm wrong. I had the feeling last IT-meet that we were telling Elmar that he was wrong. We cannot solve this by sending messages, because I feel not taken seriously and this makes me want to send angry stuff. Please, Josh, give us the time to explain our side and really try to listen and understand before trying to fix. This would also create more trust in the team.

Elmar said to me (and gave me the permission to tell this to you) that specifically the IT broke him. And this is also what I saw in his eyes. I know this cannot be what any of us want, so let's listen, understand, and fix. I proposed a fix, but probably we can find a even better fix together.

---
@WafflesMcDuff Apr 5, 2019 at 4:12 AM
I understand where you’re coming from. And I sympathize with the SLs. It’s just as painful to me when we are down for whatever reason. Thankfully, this event we were only really down for about 2 hours around breakfast on Saturday. Once we manually set the host file on all the Frontier machines, everything was accessible again including EMS in the armoury and the name generator and everything else.

The internet at the new location is much more reliable. And I’ve renewed our DNS through 2021 so this will never happen again. We already have plans (and approved budget) to correct the signal issues we experienced in the hoofdzaal, CIC and the douane area by adding additional Orbi satellites. Unfortunately, in the case of what we experienced on this event, a local-first model wouldn’t have helped us. Because we do not have admin access to the location’s WiFi, we cannot override their normal DNS settings, so running our own DNS pointing to the VM running in hyper-v wouldn’t have helped. We did have a local copy of everything running in the CiC but the failover mechanism was to re-point our DNS, which as you know was unavailable. One possible solution to this is to join all our laptops to the eosfrontier active directory domain and thus using AD-integrated DNS becomes possible for much of our architecture. However I’m loathe to do that because then NEXUS becomes a single point of failure.

Right now we have many techbare systems which are hosted in google forms and such which makes them impossible to safety-net. So rather than focusing on re-architecting everything we’ve built so far, I think we need to spend considerable focus on getting those systems to where they need to be. We can discuss this in greater detail and break it down into each of the components and suggested strategies when I’m back from Italy

---
@goblinbot Apr 4, 2019 at 6:43 AM
I personally really want to vouch for running applications like EMS, the NameGenerator and other Angular/Node applications Local-First for multiple reasons. I will expand on this with reasons/details/etc in a week.
